### PR TITLE
Job selector implemented.

### LINKED ISSUE #310: support xhtml selector

### PR DESCRIPTION
**Pull request by ashish10677**
A new parameter has been added while creating a job (`selectorType`).

Made changes to all the tests.

Closes #310

### CODE REVIEW

File: contracts/Core/AssetManager.sol [hunk 1]
```diff
@@ -10,9 +10,19 @@ import "./ACL.sol";
 contract AssetManager is ACL, AssetStorage, Constants, StateManager {
     IParameters public parameters;
 
-    event JobCreated(uint8 id, int8 power, address creator, uint32 epoch, uint256 timestamp, string name, string selector, string url);
+    event JobCreated(
+        uint8 id,
+        uint8 selectorType,
```
Comment (hrishikeshio) [inline contracts/Core/AssetManager.sol:15]: Using enum can make it more readable.

**Comment by ashish10677** [comment]
Will be fixing the VoteManager tests in the next commit.

**Review by SkandaBhat — CHANGES_REQUESTED** [review]
Please change number to enum for readability.

**Review by SkandaBhat — APPROVED** [review]
Looks good to me.

**Comment by hrishikeshio** [comment]
@abhishekvispute Can you check if variablesa re tightly packed?

**Comment by abhishekvispute** [comment]
> @abhishekvispute Can you check if variables are tightly packed?

sure

**Comment by abhishekvispute** [comment]
> @abhishekvispute Can you check if variables are tightly packed?

Yep they are 
As the number of members is 2 in this enum, it will be converted to uint8 so 
```
 event JobCreated(
        uint8 id,
        JobSelectorType selectorType,
        int8 power,
        address creator, 
        uint32 epoch, 1 slot
        uint256 timestamp, 1 slot
        string name,
        string selector,
        string url
    );
```
```
 event JobUpdated(
 uint8 id, 
 JobSelectorType selectorType, 
 uint32 epoch, 
 int8 power, 1 slot
 uint256 timestamp, 1 slot
 string selector, string URL);
 ```
```
 struct Job {
        bool active;
        uint8 id;
        uint8 assetType;
        uint8 selectorType;
        int8 power;
        uint32 epoch;
        address creator; 1 slot
        string name;
        string selector;
        string url;
    }
```

**Review by abhishekvispute — APPROVED** [review]

**Pull request merged**